Mathlib/Geometry/Manifold/IntegralCurve/UniformTime.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2023 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import Mathlib.Geometry.Manifold.IntegralCurve.ExistUnique
+
+/-!
+# Uniform time lemma for the global existence integral curves
+
+## Main results
+
+* `exists_isIntegralCurve_of_isIntegralCurveOn`: If there exists `ε > 0` such that the local
+integral curve at each point `x : M` is defined at least on an open interval `Ioo (-ε) ε`, then
+every point on `M` has a global integral curve passing through it.
+
+## Reference
+
+* [Lee, J. M. (2012). _Introduction to Smooth Manifolds_. Springer New York.][lee2012]
+
+## Tags
+
+integral curve, vector field, global existence
+-/
+
+open scoped Topology
+
+open Function Set
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners ℝ E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] [SmoothManifoldWithCorners I M] [T2Space M]
+  {γ γ' : ℝ → M} {v : (x : M) → TangentSpace I x} {s s' : Set ℝ} {t₀ : ℝ}
+
+/-- This is the uniqueness theorem of integral curves applied to a real-indexed family of integral
+  curves with the same starting point. -/
+lemma eqOn_of_isIntegralCurveOn_Ioo [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x ↦ (⟨x, v x⟩ : TangentBundle I M))) {x : M}
+    (γ : ℝ → ℝ → M) (hγx : ∀ a, γ a 0 = x) (hγ : ∀ a, IsIntegralCurveOn (γ a) v (Ioo (-a) a))
+    {a a' : ℝ} (hpos : 0 < a') (hle : a' ≤ a) :
+    EqOn (γ a') (γ a) (Ioo (-a') a') := by
+  apply isIntegralCurveOn_Ioo_eqOn_of_contMDiff_boundaryless _ hv (hγ a') ((hγ a).mono _)
+    (by rw [hγx a, hγx a'])
+  · rw [mem_Ioo]
+    exact ⟨neg_lt_zero.mpr hpos, by positivity⟩
+  · apply Ioo_subset_Ioo <;> linarith
+
+/-- Suppose for every `a : ℝ`, there exists an integral curve `γ a` on `Ioo (-a) a`, all with the
+  same starting point `γ 0 = x`. Then, the global curve `γ_ext := fun t ↦ γ (|t| + 1) t` agrees with
+  each `γ a` on `Ioo (-a) a`. This will help us show that `γ_ext` is a global integral curve. -/
+lemma exists_integralCurve_of_exists_isIntegralCurveOn_Ioo_eqOn_aux [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x ↦ (⟨x, v x⟩ : TangentBundle I M))) {x : M}
+    (h : ∀ a, ∃ γ, γ 0 = x ∧ IsIntegralCurveOn γ v (Ioo (-a) a)) {a : ℝ} :
+    EqOn (fun t' ↦ Classical.choose (h (|t'| + 1)) t') (Classical.choose (h a)) (Ioo (-a) a) := by
+  let γ := fun a' ↦ Classical.choose (h a')
+  have hγx := fun a' ↦ (Classical.choose_spec (h a')).1
```

Comment:
Can you simplify these lines using the `choose` tactic? Something like `choose γ hγx hγ using h a'` (untested).

Reply:
(Might become obsolete, depending on your response to the comment above.)